samsung-galaxybook.c

```diff
@@ -254,11 +256,15 @@ struct sawb {
 #define KBD_BACKLIGHT_MAX_BRIGHTNESS  3
 
 #define ACPI_NOTIFY_BATTERY_STATE_CHANGED    0x61
+#define ACPI_NOTIFY_TABLE_OFF                0x6d
+#define ACPI_NOTIFY_TABLE_ON                 0x6c
 #define ACPI_NOTIFY_HOTKEY_PERFORMANCE_MODE  0x70
 
 static const struct key_entry galaxybook_acpi_keymap[] = {
 	{KE_KEY, ACPI_NOTIFY_BATTERY_STATE_CHANGED, { KEY_BATTERY } },
 	{KE_KEY, ACPI_NOTIFY_HOTKEY_PERFORMANCE_MODE, { KEY_PROG3 } },
+	{KE_KEY, ACPI_NOTIFY_TABLE_OFF, { KEY_PROG1 } },
+	{KE_KEY, ACPI_NOTIFY_TABLE_ON, { KEY_PROG2 } },
```

Comment:
I forgot about this! So I wanted to add the `TABLE_ON` and `TABLE_OFF` notifications. My main concern is that I didn't find any keycode that seemed suitable to that purpose, so I chose KEY_PROG1 and KEY_PROG2, which as I understand it are explicitly generic. 

Is it reasonable to you or should we do something else?

Reply:
On this one I sort of tried to control against what seems to be typically mapped in https://github.com/systemd/systemd/blob/main/hwdb.d/60-keyboard.hwdb and in various sparse keymaps that already exist in the kernel.

As I can see it, Prog1 seems to often be used as mapping for "settings", Prog2 is a little random, Prog3 seems to often be used for "performance mode" / fan speed / etc (which is why I mapped it here) and then Prog4 plus the rest of the XF86Launch4, 5, 6+ stuff is sort of random as well.

The "downside" with this is that when using the sparse keymap it seems like they are sort of "system-wide" input events from what I can see (for example you can create keyboard shortcuts in GNOME with them and they come as the same kind of events from a regular keyboard).  So I would worry about some confusion with using Prog1 and potentially Prog2 with such specifically custom functionality. 

My two thoughts would be either 1) set them both to `KEY_UNKNOWN` and then they can be picked up using things like `evtest` and if anyone really wanted to map them to events then they would have the capability to use `setkeycodes` or similar to remap to whatever they wanted, or 2) set them to something like KEY_F14 and KEY_F15 so they can then be mapped into GNOME with "Launch5", "Launch6" etc , so they kind of fit with other "weird" stuff in systemd's hwdb plus other existing sparse keymaps that also exist in the kernel for this kind of "weird" stuff 

What do you think of this??
- either use KEY_UNKNOWN so they can be seen as input events that can then be remapped on a per-user basis , vs 
- using F14 + F15 so they can be directly used but sit sort of out of the range other commonly used mappings like PROG1 = settings etc ?

Regarding the internal names I was wondering maybe like `ACPI_NOTIFY_DEVICE_OFF_TABLE` vs  `ACPI_NOTIFY_DEVICE_ON_TABLE` as it hopefully describes it in a very clear way or do you have any objection to something like this? :)